WalletWasabi/Tor/Control/TorControlClient.cs

```diff
@@ -127,20 +153,27 @@ public Task<TorControlReply> SignalShutdownAsync(CancellationToken cancellationT
 		public Task<TorControlReply> ResetOwningControllerProcessConfAsync(CancellationToken cancellationToken = default)
 			=> SendCommandAsync("RESETCONF __OwningControllerProcess\r\n", cancellationToken);
 
-		/// <summary>
-		/// Sends a command to Tor.
-		/// </summary>
+		/// <summary>Sends a command to Tor.</summary>
 		/// <remarks>This is meant as a low-level API method, if needed for some reason.</remarks>
 		/// <param name="command">A Tor control command which must end with <c>\r\n</c>.</param>
 		public async Task<TorControlReply> SendCommandAsync(string command, CancellationToken cancellationToken = default)
 		{
-			using var _ = await MessageLock.LockAsync(cancellationToken).ConfigureAwait(false);
+			using IDisposable _ = await MessageLock.LockAsync(cancellationToken).ConfigureAwait(false);
+			return await SendCommandNoLockAsync(command, cancellationToken).ConfigureAwait(false);
+		}
+
+		/// <remarks>Lock <see cref="MessageLock"/> must be acquired by the caller.</remarks>
+		private async Task<TorControlReply> SendCommandNoLockAsync(string command, CancellationToken cancellationToken = default)
+		{
+			using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(ReaderCts.Token, cancellationToken);
 
 			Logger.LogTrace($"Client: About to send command: '{command.TrimEnd()}'");
-			await PipeWriter.WriteAsync(new ReadOnlyMemory<byte>(Encoding.ASCII.GetBytes(command)), cancellationToken).ConfigureAwait(false);
-			await PipeWriter.FlushAsync(cancellationToken).ConfigureAwait(false);
+			_ = await PipeWriter.WriteAsync(new ReadOnlyMemory<byte>(Encoding.ASCII.GetBytes(command)), linkedCts.Token).ConfigureAwait(false);
+			_ = await PipeWriter.FlushAsync(linkedCts.Token).ConfigureAwait(false);
```

Comment:
Why need `_ =`s?

Reply:
It's not needed, it's just my MSVS always shows a suggestion to write it like this. The added value is only that when somebody reads it, it's clear that a return value was not taken into account (which may be a bug).

I can remove it if you don't like it.

Reply:
I don't like lying these around, it distracts from the point of the code. Anyhow as I'm getting older, I'm less and less concerned with the codestyle, so I'm gonna merge it. Not sure that 5 minutes you'd spend replacing it and me checking it again is worth it.